mlflow/entities/metric.py

```diff
@@ -1,15 +1,33 @@
 from mlflow.entities._mlflow_object import _MLflowObject
 from mlflow.protos.service_pb2 import Metric as ProtoMetric
 
+from .metic_value_conversion_utils import (
+    convert_metric_value_to_float_if_possible,
+    convert_metric_value_to_str_if_possible,
+)
+
 
 class Metric(_MLflowObject):
     """
     Metric object.
+
+    Args:
+        key: `str`, `pyspark.ml.param.Param`, `numpy.ndarray`, `tensorflow.Tensor` or `torch.Tensor`
```

Comment:
@ntakouris Apologies if I'm asking a question that's already been answered, but can you provide some information about the use case for having numpy arrays / tensors as metric names? This seem useful as metric values, though MLflow only supports scalar metrics; however, I don't see a clear use case for these as metric names (`key` == `name`). Also, as far as I'm aware, MLflow's Python APIs will cast metric names to strings automatically during logging, so I don't think this is strictly necessary.

Reply:
I think we shouldn't do that for metric names. For param values though, we can convert ndarrays/tensors to str though.